### ISSUE TITLE
Custom Attribute Channel to Render Channel

### ISSUE THREAD

**Issue by galagast** [ChaosGroup/vray-for-houdini#84]
I currently have this simple setup:
![cust_channels_viewport](https://cloud.githubusercontent.com/assets/16534677/15600228/452297a2-241b-11e6-9cd2-263673719060.JPG)
It's a simlpe scene containing 3 objects.
![cust_channels_sep_network](https://cloud.githubusercontent.com/assets/16534677/15600254/6f88320e-241b-11e6-87ce-b9d4362ca2a9.JPG)
- Each Box object is setup to have a color attribute "Cd"
- The Floor object has a custom noise attribute "myNoise"

I then render it out with Render Channels.
Setup looks like this:
![cust_channels_ren_chan_1](https://cloud.githubusercontent.com/assets/16534677/15600327/d580f640-241b-11e6-98c0-e156955e5510.JPG)
I'm using the `TexMeshVertexColorChannel VOP` connected to a `RenderChannelExtraTexture VOP`.

Inside the Mesh Vertex Color settings, I leave it as default:
![cust_channels_mesh_vert_default](https://cloud.githubusercontent.com/assets/16534677/15600429/618bc584-241c-11e6-857c-ab3172a6a7de.JPG)

The render looks like this.. (looking at the Render Channel 'extratex')
![cust_channels_render_1](https://cloud.githubusercontent.com/assets/16534677/15600473/a144ab64-241c-11e6-830f-ad08311c163b.JPG)

It was great to see that VRay can read custom attributes!

But then I wanted to do separate render passes for each of the two attributes (Cd & myNoise). So I proceeded to set the Render Channels like so:
![cust_channels_ren_chan_2](https://cloud.githubusercontent.com/assets/16534677/15600582/79c71a08-241d-11e6-98c2-2cb59c379b02.JPG)
I wrote the attribute name inside the Channel Name parameter.

The results was not what I was expecting:
![cust_channels_render_2](https://cloud.githubusercontent.com/assets/16534677/15600644/cc8b296e-241d-11e6-8406-c6e836392579.JPG)
Both the Cd & myNoise channels showed the same rendered result. I was hoping to see the Cd channel to only show the colored Box objects (because they are the only ones that have a Cd attribute). The myNoise channel is also showing the colored Boxes, where I was hoping to see only the Floor object.

I even tried mixing up the settings and found out the ff:
- Empty the Channel Name, and set Channel Index greater than 0.. It seems that Channel Index will read any custom renderable attribute by some sort of indexing found in the object's attribute list. So if I have multiple attributes available in the object, I can pick one of those by index number. It also renders black if the index is out of range.
- Write any arbitrary random Channel Name,.. it would render out the first indexed attribute found.
  I suppose these are fine as long as the user understands what is happening.

In conclusion, it looks like these separate objects will always get rendered together because VRay will always try to look at the first existing channel. (Cd is index 0, and myNoise is also index 0)
So I tried a different approach.. I combined everything inside a single Geometry object.
![cust_channels_sop_network_2](https://cloud.githubusercontent.com/assets/16534677/15601142/a6a62a34-2420-11e6-90e8-b739e204dc9b.JPG)
Now one single object contains all the attributes I want (I even added another extra noise attribute).
![cust_channels_attr](https://cloud.githubusercontent.com/assets/16534677/15601212/fe2f276a-2420-11e6-8e86-b5ecc7751f30.jpg)
This way, I could index on my attributes safely.
The render result for myNoise worked as expected:
![cust_channels_render_3](https://cloud.githubusercontent.com/assets/16534677/15601307/2e709166-2421-11e6-9d65-6d092cba8fd7.JPG)
But the Cd channel did not (but it's almost there):
![cust_channels_render_4](https://cloud.githubusercontent.com/assets/16534677/15601349/7225843e-2421-11e6-8d16-ebfeb62fc9e3.JPG)
The only problem there is that the floor was rendered white, and not black. As a workaround, I refrained from using the Cd channel and instead created another custom attribute for that.

I was hoping that a simpler solution could be made maybe later down the development road.
Maybe just a single node to specify the attribute name you want rendered, plus maybe a control for quantization (16bit or 32bit).
Lastly, I shouldn't need to combine the objects so that I could separate the render channels during rendering. If Exclude Lists worked, that could also be a workaround for this.
These are just some thoughts. Currently the system is workable, so I guess focus on other areas of VFH could take priority for now instead of fixing this.

As a side note.. Houdini approaches it via Image Planes in Mantra, here are some other parameters they expose to control those in case anyone needs a reference.
![cust_channels_image_planes](https://cloud.githubusercontent.com/assets/16534677/15601470/1a885a34-2422-11e6-92c6-056421cf86a7.JPG)

**Comment by galagast** [comment]
The `Mesh Vertex Channel VOP` now seems to be almost working! 👍 

To be specific, I could use the custom attribute names I created in sops, and it will now be read correctly in the render channels.

Previously, I had to resort to combining the mesh as a workaround of sorts.

The only problem left.. If I specify a channel name that does not exist on a surface, it will not render using the Default Color.

[channel_name.zip](https://github.com/ChaosGroup/vray-for-houdini/files/633471/channel_name.zip)
